src/renderer.cpp

```diff
@@ -705,8 +705,8 @@ void Renderer::bindTextures()
                             if (internalState.bDoMirrorTextureWrap) flags |= BGFX_SAMPLER_U_MIRROR | BGFX_SAMPLER_V_MIRROR | BGFX_SAMPLER_W_CLAMP;
 
                             if (internalState.bIsMovie) flags |= BGFX_SAMPLER_U_CLAMP | BGFX_SAMPLER_V_CLAMP | BGFX_SAMPLER_W_CLAMP;
-
-                            if (!internalState.bDoTextureFiltering || !internalState.bIsExternalTexture) flags |= BGFX_SAMPLER_MIN_POINT | BGFX_SAMPLER_MAG_POINT | BGFX_SAMPLER_MIP_POINT;
+                            
+                          if (!internalState.bDoTextureFiltering || (!internalState.bIsExternalTexture && !filter_internal_textures)) flags |= BGFX_SAMPLER_MIN_POINT | BGFX_SAMPLER_MAG_POINT | BGFX_SAMPLER_MIP_POINT;

```

Comment:
No external flags will ever be allowed to mess with the `renderer.cpp`. Please find a way to accomplish this using the renderer internal flags. Thanks

Reply:
Renderer.cpp itself was what was deciding "never filter internal textures, full stop. ever", so that's why i put the code there.  Just deciding that the user doesn't get to chose if those textures are filtered or not is just as bad one way as the other.  The flag being there is exactly what i intended.

That said. I have deleted that check from renderer.cpp, and attempted to move it to the end of specialcase, because s I understand the code right, that should apply it properly.

Will compile a new one and test once more for myself.

Reply:
Okay i moved the check, but when i do so it forces filtering on the square enix logo movie, which apparently skips special_case completely somehow..

So if you can tell me how to actually catch that case properly, i'll fix the code, but i have no idea HOW.

Reply:
Left a comment on the same line down below. Closing this thread.

Reply:
yeah, i can't figure out how to even check if the movie is external there or not. the code from special_case doesn't compile.